chunkstorage.go

```diff
@@ -3,71 +3,47 @@ package desync
 import (
 	"bytes"
 	"context"
-	"crypto/sha512"
 	"fmt"
 	"os"
 	"sync"
 
-	"io"
-
 	"github.com/pkg/errors"
 )
 
-type ChunkJob struct {
-	num   int
-	chunk IndexChunk
-}
-
 type ChunkStorage struct {
 	sync.Mutex
```

Comment:
sync.RWMutex would likely a little better here to avoid blocking during concurrent reads